moto/sagemaker/responses.py

```diff
@@ -796,3 +796,174 @@ def update_endpoint_weights_and_capacities(self) -> TYPE_RESPONSE:
+        sort_by = self._get_param("SortBy")
+        sort_order = self._get_param("SortOrder")
+        (
+            model_package_summary_list,
+            next_token,
+        ) = self.sagemaker_backend.list_model_packages(
+            creation_time_after=creation_time_after,
+            creation_time_before=creation_time_before,
+            max_results=max_results,
+            name_contains=name_contains,
+            model_approval_status=model_approval_status,
+            model_package_group_name=model_package_group_name,
+            model_package_type=model_package_type,
+            next_token=next_token,
+            sort_by=sort_by,
+            sort_order=sort_order,
+        )
+        # TODO: adjust response
+        response_values = [
+            "ModelPackageName",
+            "ModelPackageGroupName",
+            "ModelPackageVersion",
+            "ModelPackageArn",
+            "ModelPackageDescription",
+            "CreationTime",
+            "ModelPackageStatus",
+            "ModelApprovalStatus",
+        ]
+        model_package_summary_list_response_object = [
+            x.gen_response_object() for x in model_package_summary_list
+        ]
+        model_package_summary_list_response_object_filtered = [
+            {key: x[key] for key in response_values}
+            for x in model_package_summary_list_response_object
+        ]
+        for response_object in model_package_summary_list_response_object_filtered:
+            response_object["CreationTime"] = response_object["CreationTime"].isoformat()
+        return json.dumps(
+            dict(
+                ModelPackageSummaryList=model_package_summary_list_response_object_filtered,
+                NextToken=next_token,
+            )
+        )
+
+    def describe_model_package(self):
+        model_package_name = self._get_param("ModelPackageName")
+        (
+            model_package_name,
+            model_package_group_name,
+            model_package_version,
+            model_package_arn,
+            model_package_description,
+            creation_time,
+            inference_specification,
+            source_algorithm_specification,
+            validation_specification,
+            model_package_status,
+            model_package_status_details,
+            certify_for_marketplace,
+            model_approval_status,
+            created_by,
+            metadata_properties,
+            model_metrics,
+            last_modified_time,
+            last_modified_by,
+            approval_description,
+            customer_metadata_properties,
+            drift_check_baselines,
+            domain,
+            task,
+            sample_payload_url,
+            additional_inference_specifications,
+        ) = self.sagemaker_backend.describe_model_package(
+            model_package_name=model_package_name,
+        )
+        # TODO: adjust response
+        return json.dumps(
+            dict(
```

Comment:
Can we use `gen_response_object` here? Ideally we avoid listing all attributes separately

Reply:
This is in the create_model_package not the described.
I think this is more suitable to not do that no ? gen_response_object should be for lists and describe but here I only need ModelPackageArn ? wdyt ?